### ISSUE TITLE
Alt text for image posts

### ISSUE THREAD

**Issue by glibg10b** [LemmyNet/lemmy-ui#2373]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

When posting an image, we have the option to add a body of text, just as with text posts:

![Screenshot_20240222_212213_Firefox Nightly](https://github.com/LemmyNet/lemmy-ui/assets/56197853/b11fb2b0-b9ce-4ff0-97c9-a1c8f92da6fa)

However, unlike with Markdown images (`![alt text]()`), there's no option to add alt text. This leads to everyone using the body text for image transcriptions, which makes this text visible to everyone:

![Screenshot_20240222_211901_Firefox Nightly](https://github.com/LemmyNet/lemmy-ui/assets/56197853/10b67ab9-0d38-4da1-9e7d-fd5144566eb5)

I suggest that a separate "image transcription [optional]" field be added to the post creation form.

**Comment by travis-jeans** [comment]
A few reasons why I think the image doesn't have alt text:


## Image Links

The images (image thumbnail, image in post) are links and alt text for links should be the destination of the link, see [H30: Providing link text that describes the purpose of a link for anchor elements (WCAG)](https://www.w3.org/WAI/WCAG22/Techniques/html/H30). Based on this:


## Thumbnail Image Button

Currently the thumbnail image on Image Posts is marked as decorative with blank alt text `alt=""` and the button surrounding it has `aria-label="Expand here"` which is suitable since it describes the button. `<button class="thumbnail rounded overflow-hidden d-inline-block position-relative p-0 border-0 bg-transparent" type="button" data-tippy-content="Expand here" aria-label="Expand here">`


## Expanded Image Link

Once the image is expanded (whether on the post preview or in the post itself) the image is made larger and is wrapped in a link to the full size image. This might be a result of the image being inserted with Markdown since the default behavior is to provide links to the full size image in Markdown. However it would be better if there was a way to add alt text to this expanded image so I have added some suggestions below.


## Possible solutions

- when the image is expanded, it is expanded to a set height / width as determined by Markdown but the link to the full size image is removed. Instead, the link to the full size image is added after with text (Full size image (width x height, image size kB/MB)) or something like that, while user made alt text can be added to the image itself.
- alternatively, the expanded image is still a link to a full size image with the link accessible title or alt text as  "full size image" so screenreader users understand the link is for the full size image. Then user-inserted alt text / image description is added to a display/hide widget underneath that can be toggled open and is written in Markdown

**Comment by dessalines** [comment]
This is already done for a few versions now.
